spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkMicroBatchStream.java

```diff
@@ -111,7 +112,10 @@ public Offset latestOffset() {
     }
 
     Snapshot latestSnapshot = table.currentSnapshot();
-    return new StreamingOffset(latestSnapshot.snapshotId(), Iterables.size(latestSnapshot.addedFiles()), false);
+    return new StreamingOffset(latestSnapshot.snapshotId(),
+        Long.parseLong(latestSnapshot.summary().getOrDefault(SnapshotSummary.ADDED_FILES_PROP,
```

Comment:
can use `PropertyUtil.propertyAsLong`. Also this is too long to squeeze in a single line call. Can we have 1 line getting the added files, another line to call `return new StreamingOffset(latestSnapshot.snapshotId(), addedFilesCount, false);`

Reply:
I see, I didn't use `PropertyUtil.propertyAsLong` as to provide the default, I will have to iterate the addedFiles iterator to compute it's size (implying materilizing this), presently this util doesn't supports sending it a supplier so that it can be computed only when key is not present. what do you recommend here ?

Reply:
I also see the similar problem will be there in my current approach in the way hashMap is implemented it. 
```
    default V getOrDefault(Object key, V defaultValue) {
        V v;
        return (((v = get(key)) != null) || containsKey(key))
            ? v
            : defaultValue;
    }
```

Let me try to fix it